userspace/otpilot/src/main.rs

```diff
@@ -160,73 +163,113 @@ impl rsa::Builder for NoRsa {
 
 //////////////////////////////////////////////////////////////////////////////
 
+#[derive(Copy, Clone, Debug)]
+enum SpiProcessorError {
+    FromWire(FromWireError),
+    ToWire(ToWireError),
+    Tock,
+    Manticore(manticore::server::Error),
+    UnsupportedContentType(payload::ContentType),
+    UnsupportedOpCode(OpCode),
+    InvalidAddress(Option<u32>),
+    Format(core::fmt::Error),
+}
+
+impl From<FromWireError> for SpiProcessorError {
+    fn from(err: FromWireError) -> Self {
+        SpiProcessorError::FromWire(err)
+    }
+}
+
+impl From<core::num::TryFromIntError> for SpiProcessorError {
```

Comment:
I'm not sure I like the use of `From<core::num::TryFromIntError>` to create a `SpiProcessorError::FromWire`. It doesn't feel like a clean abstraction. As far as I can tell, it is only used in one location.

Perhaps we should remove this `From` implementation and replace `u16::try_from(tx_cursor.consumed_len())?` with `u16::try_from(tx_cursor.consumed_len()).map_err(|_| FromWireError::OutOfRange.into())?`

Reply:
Good point. The other cases should be clean. Thanks for pointing this out!